### PR TITLE
fix: registry column keyerror + registry df + remote path bugs

### PR DESCRIPTION
**Pull request by r614**
- Registry records should not raise KeyError when accessing a column with Optional Value 
- Added `Table.get_dataframe` 
- Added nicer paths for LatchFile and LatchDir

### CODE REVIEW

File: latch/registry/table.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import gql
 import graphql.language as l
 import graphql.language.parser as lp
+import pandas as pd
```
Comment (ayushkamat) [inline latch/registry/table.py:24]: we should make pandas an optional dep and if people want this functionality they can opt in by doing `pip install latch[pandas]` or smth

File: latch/registry/table.py [hunk 2]
```diff
@@ -265,6 +266,42 @@ def list_records(self, *, page_size: int = 100) -> Iterator[Dict[str, Record]]:
...
+
+    @overload
+    def get_dataframe(self, *, load_if_missing: bool) -> Optional[pd.DataFrame]:
+        ...
+
+    def get_dataframe(self, *, load_if_missing: bool = True) -> Optional[pd.DataFrame]:
+        """Get a pandas DataFrame of all records in this table.
+
+        Returns:
+            DataFrame reperesenting all records in this table.
```
Comment (ayushkamat) [inline latch/registry/table.py:283]:
```suggestion
            DataFrame representing all records in this table.
```

File: latch/registry/table.py [hunk 3]
```diff
@@ -265,6 +266,42 @@ def list_records(self, *, page_size: int = 100) -> Iterator[Dict[str, Record]]:
         if len(page) > 0:
             yield page
 
+    @overload
+    def get_dataframe(
+        self, *, load_if_missing: Literal[True] = True
+    ) -> Optional[pd.DataFrame]:
```
Comment (ayushkamat) [inline latch/registry/table.py:272]:
```suggestion
    ) -> pd.DataFrame:
```

Comment (ayushkamat) [reply latch/registry/table.py:272]: since when load_if_missing is true its impossible to get a null return

Comment (ayushkamat) [reply latch/registry/table.py:272]: also idt it makes sense to have a `load_if_missing` param here, i don't see the utility of passing in `load_if_missing=False`

File: latch/registry/table.py [hunk 4]
```diff
@@ -265,6 +266,42 @@ def list_records(self, *, page_size: int = 100) -> Iterator[Dict[str, Record]]:
...
+                if full_record is not None:
+                    full_record["Sample_Name"] = record.get_name(
+                        load_if_missing=load_if_missing
+                    )
+                    records.append(full_record)
+
+        if len(records) == 0:
+            cols = self.get_columns(load_if_missing=load_if_missing)
+            if cols is None:
+                return None
```
Comment (ayushkamat) [inline latch/registry/table.py:299]: return an empty dataframe instead

File: latch/registry/table.py [hunk 5]
```diff
@@ -265,6 +266,42 @@ def list_records(self, *, page_size: int = 100) -> Iterator[Dict[str, Record]]:
...
+        Returns:
+            DataFrame reperesenting all records in this table.
+        """
+
+        records = []
+        for page in self.list_records():
+            for record_id, record in page.items():
+                full_record = record.get_values(load_if_missing=load_if_missing)
+                if full_record is not None:
+                    full_record["Sample_Name"] = record.get_name(
```
Comment (ayushkamat) [inline latch/registry/table.py:291]: why `Sample_Name` and not `Name` or `Record_Name`

File: latch/types/directory.py [hunk 6]
```diff
@@ -182,13 +182,17 @@ def remote_path(self) -> Optional[str]:
 
     def __repr__(self):
         if self.remote_path is None:
-            return f'LatchDir("{self.local_path}")'
-        return f'LatchDir("{self.path}", remote_path="{self.remote_path}")'
+            formatted_path = format_path(self.local_path)
+            return f'LatchDir("{formatted_path}")'
+
+        formatted_path = format_path(self.remote_path)
+        return f'LatchDir("{self.path}", remote_path="{formatted_path}")'
```
Comment (ayushkamat) [inline latch/types/directory.py:189]: id inline the fn calls but idrc that much

File: latch/types/utils.py [hunk 7]
```diff
@@ -17,3 +21,44 @@ def _is_valid_url(raw_url: Union[str, Path]) -> bool:
...
     return True
+
+
+is_absolute_node_path = re.compile(r"^(latch)?://(?P<node_id>\d+).node(/)?$")
+
+
+def format_path(path: str) -> str:
+    match = is_absolute_node_path.match(path)
+
+    if not match:
```
Comment (ayushkamat) [inline latch/types/utils.py:32]:
```suggestion
    if match is None:
```

File: latch/types/utils.py [hunk 8]
```diff
@@ -17,3 +21,44 @@ def _is_valid_url(raw_url: Union[str, Path]) -> bool:
...
+    path_split = raw_path.split("/")
+
+    if path_split[0] == "mount":
+        mount_name = path_split[1]
+        fpath = "/".join(path_split[2:])
+        return f"latch://{mount_name}.mount/{fpath}"
+
+    if path_split[0] == "account_root":
+        fpath = "/".join(path_split[2:])
+        return f"latch://{owner}.account/{fpath}"
```
Comment (ayushkamat) [inline latch/types/utils.py:62]: id rather this be done with a regex but nbd for now

File: requirements.in [hunk 9]
```diff
@@ -20,3 +20,4 @@ asyncssh==2.12.0
 watchfiles==0.18.1
 gql==3.4.0
 aiohttp==3.8.4
+pandas==1.3.5
```
Comment (ayushkamat) [inline requirements.in:23]: make this an optional dep in the setup.py file

**Review by ayushkamat — CHANGES_REQUESTED** [review]

File: latch/registry/table.py (hunk 1 above)
Comment (r614) [reply latch/registry/table.py:24]: pandas is pulled in by flytekit anyways, should probably fix that upstream

Comment (ayushkamat) [reply latch/registry/table.py:24]: yee

**Pull request merged**